conda_build/metadata.py

```diff
@@ -2641,12 +2641,13 @@ def get_loop_vars(self):
         return variants.get_vars(_variants, loop_only=True)
 
     def get_used_loop_vars(self, force_top_level=False, force_global=False):
+        loop_vars = set(self.get_loop_vars())
         return {
             var
             for var in self.get_used_vars(
                 force_top_level=force_top_level, force_global=force_global
             )
-            if var in self.get_loop_vars()
+            if var in loop_vars
         }
```

Comment:
Isn't this just a set operation?

```python
used_vars = set(self.get_used_vars(...))
loop_vars = set(self.get_loop_vars())
return used_vars & loop_vars
```

Reply:
Certainly! The changes from gh-5225 (and thus the once cherry-picked here) were the minimal/"just-bolted-on" changes with no drive-by cleanups.
But we can, of course, apply these cleanups here too if we touch these lines anyway.

Reply:
I put this now in as `return set(loop_vars).intersection(used_vars)` -- in this case here it probably doesn't make much of a difference, but if the second iterable (`used_vars`) is large, then `set.intersection` can be a bit faster since one avoids the creation of the intermediate `set` (IIRC, `a & b` also runs `a.intersection(b)` internally, but just with additional type checks -- but `set.intersection` takes iterables as inputs anyway).